src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.ContentSearch/SearchResultsController.ts

```diff
@@ -33,21 +35,64 @@ module Orckestra.Composer {
                     TotalCount: Total,
                     SelectedSortBy,
                     AvailableSortBys,
+                    FacetsVisible: FacetsVisible,
                     Pagination: {
                         PagesCount: 1,
                         CurrentPage: 1,
                         PreviousPage: false,
                         NextPage: false,
                     },
+                    isOverriden: isOverriden,
                     isLoading: false
                 },
                 mounted() {
                     this.Pagination = this.getPagination(PagesCount)
                     self.eventHub.subscribe(ContentSearchEvents.SearchResultsLoaded, this.onSearchResultsLoaded);
+                    
+                    if (FacetsVisible == "") {
+                        this.context.container.data('facets-visible', "True");
+                        self.vueSearchResults.$el.setAttribute('data-facets-visible', 'True')
+                        FacetsVisible = "True";
+                        this.vueSearchResults.data.FacetsVisible = "True";
+                    }
                 },
                 computed: {
                 },
+                updated: function () {
+                    this.updateProductColumns();
+                },
                 methods: {
+                    toggleFacet(): void {
+                        let rightCol = document.getElementById("rightCol");
+                        if (FacetsVisible == "True") {
+                            FacetsVisible = "False";
+                            self.vueSearchResults.$el.setAttribute('data-facets-visible', 'False');
+                            document.getElementById("leftCol").classList.add("w-0-lg");
+                            rightCol.classList.remove("col-lg-9");
+                            
+                        }
+                        else {
+                            FacetsVisible = "True";
+                            self.vueSearchResults.$el.setAttribute('data-facets-visible', 'True');
+                            document.getElementById("leftCol").classList.remove("w-0-lg");
+                            rightCol.classList.add("col-lg-9");
+                        }
+                        self.vueSearchResults.$data.FacetsVisible = FacetsVisible;
```

Comment:
it is better have boolean Vue property and here have 
this.FacetsVisible  = !this.FacetsVisible;